examples/terminating-gateway-tls/efs.tf

```diff
@@ -0,0 +1,144 @@
+resource "aws_security_group" "efs" {
+  name        = "efs-sg"
+  description = "Allows inbound efs traffic from ec2"
+  vpc_id      = module.vpc.vpc_id
+
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_efs_file_system" "certs_efs" {
+  creation_token   = "certs-efs"
+  performance_mode = "generalPurpose"
+  throughput_mode  = "bursting"
+  encrypted        = "true"
+  tags = {
+    Name = "Certs"
+  }
+}
+
+
+resource "aws_efs_mount_target" "efs_mt" {
+  count           = length(module.vpc.private_subnets)
+  file_system_id  = aws_efs_file_system.certs_efs.id
+  subnet_id       = module.vpc.private_subnets[count.index]
+  security_groups = [aws_security_group.efs.id]
+}
+
+#############################################################
+
+# generate ca cert and key
+resource "tls_private_key" "tgw_external_app_ca_key" {
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P256"
+}
+
+resource "tls_self_signed_cert" "tgw_external_app_ca_cert" {
+  private_key_pem       = tls_private_key.tgw_external_app_ca_key.private_key_pem
+  validity_period_hours = 43800
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+    "cert_signing",
+    "crl_signing",
+  ]
+  dns_names = ["*"]
+  subject {
+    common_name  = "*"
+    organization = "HashiCorp Inc."
+  }
+
+  is_ca_certificate  = true
+  set_subject_key_id = true
+}
+
+resource "aws_secretsmanager_secret" "tgw_external_app_ca_key" {
+  name                    = "tgw-external-app-ca-key"
+  recovery_window_in_days = 0
+}
+
+resource "aws_secretsmanager_secret_version" "tgw_external_app_ca_key" {
+  secret_id     = aws_secretsmanager_secret.tgw_external_app_ca_key.id
+  secret_string = tls_private_key.tgw_external_app_ca_key.private_key_pem
+}
+
+resource "aws_secretsmanager_secret" "tgw_external_app_ca_cert" {
+  name                    = "tgw-external-app-ca-cert"
+  recovery_window_in_days = 0
+}
+
+resource "aws_secretsmanager_secret_version" "tgw_external_app_ca_cert" {
+  secret_id     = aws_secretsmanager_secret.tgw_external_app_ca_cert.id
+  secret_string = tls_self_signed_cert.tgw_external_app_ca_cert.cert_pem
+}
+
+#generate cert and key for the gateway
+resource "tls_private_key" "tgw_external_app_private_key" {
```

Comment:
Similar to the previous comment, I would suggest to name this only `gateway_private_key` or something similar. The idea is to not club TGW and external app in a single name because they are two different entities.